app/src/pages/inside/dashboardItemPage/widgetWizardModal/widgetControls/controls/filtersControl/filterAdd/filterAddFormBlock/filterAddFormBlock.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import { defineMessages, intlShape, injectIntl } from 'react-intl';
+import classNames from 'classnames/bind';
+import { reduxForm } from 'redux-form';
+import { validate } from 'common/utils';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { Input } from 'components/inputs/input';
+import { FILTER_ADD_FORM, FILTER_NAME_KEY } from '../../common/constants';
+import styles from './filterAddFormBlock.scss';
+
+const cx = classNames.bind(styles);
+const messages = defineMessages({
+  filterName: {
+    id: 'AddFilter.filterName',
+    defaultMessage: 'Filter Name',
+  },
+  placeholderFilterName: {
+    id: 'AddFilter.placeholderFilterName',
+    defaultMessage: 'Input filter name',
+  },
+});
+
+@reduxForm({
+  form: FILTER_ADD_FORM,
+  validate: ({ name }) => ({
+    name: (!name || !validate.filterName(name)) && 'filterNameError',
+  }),
+  onChange: (fields, dispatcher, { onChange }) => onChange(),
```

Comment:
Maybe it's better to move redux-form upper? Triggering empty `onChange` to trigger something on the upper level (where you still use this new value, but get it in the other way) looks very strange.

Reply:
Has been rewritten.